R/aorsf_data.R

```diff
@@ -0,0 +1,225 @@
+# nocov start
+
+make_rand_forest_aorsf <- function(){
+
+
+  parsnip::set_model_engine("rand_forest", "classification", "aorsf")
+  parsnip::set_model_engine("rand_forest", "regression", "aorsf")
+  parsnip::set_dependency("rand_forest", "aorsf", "aorsf", mode = "classification")
+  parsnip::set_dependency("rand_forest", "aorsf", "aorsf", mode = "regression")
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "mtry",
+    original = "mtry",
+    func = list(pkg = "dials", fun = "mtry"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "trees",
+    original = "n_tree",
+    func = list(pkg = "dials", fun = "trees"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "min_n",
+    original = "leaf_min_obs",
+    func = list(pkg = "dials", fun = "min_n"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "mtry",
+    original = "mtry",
+    func = list(pkg = "dials", fun = "mtry"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_fit(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    value = list(
+      interface = "formula",
+      protect = c("formula", "data", "weights"),
+      func = c(pkg = "aorsf", fun = "orsf"),
+      defaults =
+        list(
+          n_thread = 1,
+          verbose_progress = FALSE
+        )
+    )
+  )
+
+  parsnip::set_encoding(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    options = list(
+      predictor_indicators = "none",
+      compute_intercept = FALSE,
+      remove_intercept = FALSE,
+      allow_sparse_x = FALSE
+    )
+  )
+
+  parsnip::set_fit(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "regression",
+    value = list(
+      interface = "formula",
+      protect = c("formula", "data", "weights"),
+      func = c(pkg = "aorsf", fun = "orsf"),
+      defaults =
+        list(
+          n_thread = 1,
+          verbose_progress = FALSE
+        )
+    )
+  )
+
+  parsnip::set_encoding(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "regression",
+    options = list(
+      predictor_indicators = "none",
+      compute_intercept = FALSE,
+      remove_intercept = FALSE,
+      allow_sparse_x = FALSE
+    )
+  )
+
+
+  parsnip::set_pred(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    type = "class",
+    value = list(
+      pre = NULL,
+      # makes prob preds consistent with class ones
```

Comment:
Could you leave a few more sentences of explanation as to why this is needed? I would have thought that, for `type = "class"`, you'd just pass `pred_type = "class"` (which is passed directly to `aorsf::predict.ObliqueForest()`) in the `args` below.

Reply:
Ah, I now see the code comment in the tests. Feel free to just move that up here and link to this PR using `(#78)` at the end of the code comment.

Reply:
Let's also add this to the classification section in the new docs at https://github.com/tidymodels/parsnip/pull/1120!

Reply:
Good call. The note should be visible in both places now. I added it to `other details` in `parsnip`'s docs.